astrorapid/prepare_training_set.py

```diff
@@ -357,10 +357,11 @@ def multi_read_obj(self, objids):
             y[i][0:len_t][activeindexes] = class_name
 
         deleterows = np.array(deleterows)
-        X = np.delete(X, deleterows, axis=0)
-        y = np.delete(y, deleterows, axis=0)
-        labels = np.delete(labels, deleterows, axis=0)
-        timesX = np.delete(timesX, deleterows, axis=0)
+        if deleterows.size > 0:
```

Comment:
@daniel-muthukrishna small bug fix here. The `np.delete` gives an `IndexError` if `deleterows` is an empty array. 
I would also be happy to refactor into a different PR if you wanted.